src/deluge/model/voice/voice.cpp

```diff
@@ -1090,49 +1092,7 @@ skipAutoRelease: {}
 	bool stereoUnison = sound->unisonStereoSpread && sound->numUnison > 1 && soundRenderingInStereo;
 
 	// If various conditions are met, we can cut a corner by rendering directly into the Sound's buffer
-	bool renderingDirectlyIntoSoundBuffer;
-
-	// Lots of conditions rule out renderingDirectlyIntoSoundBuffer right away
-	if (sound->clippingAmount
-	    || sound->synthMode == SynthMode::RINGMOD // We could make this one work - but currently the ringmod rendering
-	                                              // code doesn't really have
-	    // proper amplitude control - e.g. no increments - built in, so we rely on the normal final
-	    // buffer-copying bit for that
-	    || filterSet.isHPFOn() || filterSet.isLPFOn()
-	    || (paramFinalValues[params::LOCAL_NOISE_VOLUME] != 0
-	        && synthMode != SynthMode::FM) // Not essential, but makes life easier
-	    || paramManager->getPatchCableSet()->doesParamHaveSomethingPatchedToIt(params::LOCAL_PAN)
-	    || (paramFinalValues[params::LOCAL_FOLD] != NEGATIVE_ONE_Q31)) {
-		renderingDirectlyIntoSoundBuffer = false;
-	}
-
-	// Otherwise, we need to think about whether we're rendering the same number of channels as the Sound
-	else {
-		if (synthMode == SynthMode::SUBTRACTIVE) {
-
-			for (int32_t s = 0; s < kNumSources; s++) {
-				if (!sound->isSourceActiveCurrently(s, paramManager)) {
-					continue;
-				}
-
-				bool renderingSourceInStereo =
-				    sound->sources[s].renderInStereo(sound, (SampleHolder*)guides[s].audioFileHolder);
-
-				if (renderingSourceInStereo != soundRenderingInStereo) {
-					renderingDirectlyIntoSoundBuffer = false;
-					goto decidedWhichBufferRenderingInto;
-				}
-			}
-
-			// If still here, no mismatch, so go for it
-			renderingDirectlyIntoSoundBuffer = true;
-		}
-		else {
-			// If got here, we're rendering in mono
-			renderingDirectlyIntoSoundBuffer = (soundRenderingInStereo == false);
-		}
-	}
-decidedWhichBufferRenderingInto:
+	bool renderingDirectlyIntoSoundBuffer = false;
```

Comment:
I think this can be deleted? The corresponding downstream code is gone now.

Reply:
I think it's still in use, it gets passed into some of the other functions

Reply:
I could make it constexpr?

Reply:
I got it all removed, it's used less than I thought